guides/host_record.md

```diff
@@ -0,0 +1,130 @@
+
+Host Record serves as the backend for the following operations:
+
+- Allocation and deallocation of an IP address from a network (`nios_ip_allocation`)
+- Association and disassociation of an IP address with a VM (`nios_ip_association`)
+
+---
+
+## Notes
+
+- `nios_ip_allocation` **owns the Host Record lifecycle**. Create the allocation first - it creates the Host Record and DNS entries.
+- `nios_ip_association` is **read & update only** and must reference an existing allocation via `ref`. Do **not** create the association before the allocation exists.
+- Do **not** destroy the association resource directly - destroy the allocation and the provider will remove the associated Host Record and DHCP settings.
+- Each allocation supports **at most one IPv4 and one IPv6** address. You may have one of each (dual-stack), but multiple addresses of the same family are not supported.
+- Use `nios_ip_association` to attach instance identifiers (MAC / DUID) and DHCP flags only; allocation handles DNS and address allocation.
+
+### Recommended workflow
+1. Create `nios_ip_allocation` (allocates IP + manages DNS).
+2. Create `nios_ip_association` referencing allocation (`ref = nios_ip_allocation.<name>.ref`) to attach MAC/DUID and DHCP settings.
+
+### Destroy order
+- To remove a Host Record and its DHCP association:
+  - Destroy `nios_ip_allocation` (this will remove the Host Record and the association).
+  - Do NOT destroy the `nios_ip_association` first.
+
+---
+
+## Examples
+
+### Static IPv4 Host Record with MAC Address
+
+```terraform
+# Allocate a static IPv4 address with DNS configuration
+resource "nios_ip_allocation" "allocation_static" {
+  name              = "host1.example.com"
+  view              = "default"
+  configure_for_dns = true
+  ipv4addrs = [
+    {
+      ipv4addr = "10.101.1.110"
+    }
+  ]
+  extattrs = {
+    Site = "location-1"
+  }
+}
+
+# Associate MAC address without enabling DHCP
+resource "nios_ip_association" "association_static" {
+  ref                = nios_ip_allocation.allocation_static.ref
+  mac                = "12:00:43:fe:9a:8c"
+  configure_for_dhcp = false
+}
+```
+
+### Dual-Stack Host Record with DHCP Enabled
+
+```hcl
+# Allocate both IPv4 and IPv6 addresses
+resource "nios_ip_allocation" "allocation_dual_stack" {
+  name              = "host2.example.com"
+  view              = "default"
+  configure_for_dns = true
+  ipv4addrs = [
+    {
+      ipv4addr = "10.101.1.112"
+    }
+  ]
+  ipv6addrs = [
+    {
+      ipv6addr = "2002:1f93::12:2"
+    }
+  ]
+  extattrs = {
+    Site = "location-1"
+  }
+}
+
+# Associate MAC and DUID with DHCP enabled
+resource "nios_ip_association" "association2_dual_stack" {
+  ref                = nios_ip_allocation.allocation_dual_stack.ref
```

Comment:
w.r.t Add one note saying recommended to create a  implicit dependencies, to generate dependency graph properly. With this it will execute allocation first and then association. As each resource thread is independent